JoonasHeinonen/escape-from-zordoom-prison
# Feature/level selection view

### Link to the ticket: [Level selection view](https://trello.com/c/dEiCKY4i/39-level-selection-view)
![image](https://user-images.githubusercontent.com/33086486/192882798-e9847274-56e2-4d22-87ae-5ae538a5b8ff.png)

**Changes:**
- Added a new scene to list all the available scenes from the `Levels` subfolder.
- Lists all the levels in the scene as buttons.
- Implemented naming convention and string converter for the name of the levels. E.g. it adds the level name onto a button in the following way: `some_new_map` => `Some New Map`
- When clicked on a button, moves to the scene that is connected to the button.
- Added `esc`-button into the input map. It changes the current scene to the main menu.
- Created a new theme for the saved games and levels.
- Edited the existing fonts to fit into the new theme.

**How to test:**
 - While on the main menu, click the button `SELECT LEVEL`.
 - On the following scene, the current levels on the Levels-folder are listed. Click any of these. It will swap the scene to the given scene.
 - In any level that has player on it (currently, the only one having a player character is `test_level.tscn`), allow the game to load the scene, usually takes about 10 seconds. Press `esc` to return to the main menu.
 - Verify that the current styling, fading animation when choosing a level and the new theme for the saved games and levels is adequate. 

**Documentation & sources:**
 - [How to get all the files inside a folder?](https://godotengine.org/qa/5175/how-to-get-all-the-files-inside-a-folder)
 - [How to edit button text using scrpit?](https://godotengine.org/qa/81612/how-to-edit-button-text-using-scrpit)
 - [How to remove all punctuation from a string in Godot?](https://stackoverflow.com/questions/72860320/how-to-remove-all-punctuation-from-a-string-in-godot)
 - [Remove last character from variable (a string)](https://www.reddit.com/r/godot/comments/h8a2nc/remove_last_character_from_variable_a_string/)
 - [String capitalize ( )](https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-capitalize)
